packages/frontend/src/routes/CreateImplicitAccountWrapper.js

```diff
@@ -0,0 +1,131 @@
+import BN from 'bn.js';
+import { getLocation } from 'connected-react-router';
+import { utils } from 'near-api-js';
+import { formatNearAmount } from 'near-api-js/lib/utils/format';
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+import CreateImplicitAccount from '../components/accounts/create/implicit_account/CreateImplicitAccount';
+import { MIN_BALANCE_TO_CREATE } from '../config';
+import { Mixpanel } from '../mixpanel';
+import { redirectTo, checkAndHideLedgerModal } from '../redux/actions/account';
+import { showCustomAlert } from '../redux/actions/status';
+import { selectAccountId } from '../redux/slices/account';
+import { actions as createFromImplicitActions } from '../redux/slices/createFromImplicit';
+import { getSignedUrl } from '../utils/moonpay';
+import { isMoonpayAvailable } from '../utils/moonpay';
+import useRecursiveTimeout from '../utils/useRecursiveTimeout';
+import { wallet } from '../utils/wallet';
+
+const { setCreateCustomName } = createFromImplicitActions;
+
+export function CreateImplicitAccountWrapper() {
+    const dispatch = useDispatch();
+
+    const accountId = useSelector(selectAccountId);
+
+    const [fundingNeeded, setFundingNeeded] = useState(true);
+    const [moonpaySignedUrl, setMoonpaySignedUrl] = useState('');
+
+    const location = useSelector(getLocation);
+    const URLParams = new URLSearchParams(location.search);
+    const implicitAccountId = URLParams.get('implicitAccountId');
+    const recoveryMethod = URLParams.get('recoveryMethod');
+
+    const formattedMinDeposit = formatNearAmount(MIN_BALANCE_TO_CREATE);
+
+    // Check that the initial deposit was at least 0.2N, otherwise the users 'available balance'
+    // won't be enough to create a named account.
+    const NAMED_ACCOUNT_MIN = utils.format.parseNearAmount('0.2');
+
+    useEffect(() => {
+        if (accountId === implicitAccountId || !implicitAccountId || !recoveryMethod) {
+            dispatch(redirectTo('/'));
+        }
+    }, [accountId, implicitAccountId, recoveryMethod]);
+
+    useEffect(() => {
+        const checkIfMoonPayIsAvailable = async () => {
+            await Mixpanel.withTracking("CA Check Moonpay available",
+                async () => {
+                    const moonpayAvailable = await isMoonpayAvailable();
+                    if (moonpayAvailable) {
+                        const moonpaySignedUrl = await getSignedUrl(implicitAccountId, window.location.href, 30);
+                        setMoonpaySignedUrl(moonpaySignedUrl);
+                    }
+                },
+                (e) => {
+                    throw e;
+                }
```

Comment:
1. :hocho: this is a no-op (if you don't add an error handler function, the error bubbles up to the caller anyway)
2. Nothing will actually catch this error; it will bubble all the way up to the browser and be sent as a sentry exception.  Should this silently fail (wrap this in a try/catch to squelch errors), or do we want this behaviour?